src/app/components/itinerary/itinerarylist/itinerarylist.component.ts

```diff
@@ -9,26 +11,42 @@ import { ItineraryService } from 'src/app/services/itinerary.service';
 })
 export class ItinerarylistComponent implements OnInit {
 
-  constructor(private itineraryService: ItineraryService) { }
+  constructor(private itineraryService: ItineraryService,private route: ActivatedRoute, private tokenService: TokenService) { }
 
   itineraries: Itinerary[] = [];
+  numberOfElements: Number;
+  totalPages: Array<Number>=[];
+  pageId: Number;
+
+  @Input()
+  userId: Number;
+  
+  username: String;
+
 
   ngOnInit(): void {
-    this.cargarItineraries();
+    this.pageId = Number(this.route.snapshot.paramMap.get('id')) -1;
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ispp-yourney/angular-app/pullRequest?prid=7157930)
```suggestion
    this.pageId = Number(this.route.snapshot.paramMap.get("id")) -1;
```